lib/kafo/kafo_configure.rb

```diff
@@ -375,11 +399,83 @@ def unset
 
     def config_file
       return CONFIG_FILE if defined?(CONFIG_FILE) && File.exists?(CONFIG_FILE)
+      return select_scenario(CONFIG_DIR) if defined?(CONFIG_DIR) && Dir.exists?(CONFIG_DIR)
       return '/etc/kafo/kafo.yaml' if File.exists?('/etc/kafo/kafo.yaml')
       return "#{::RbConfig::CONFIG['sysconfdir']}/kafo/kafo.yaml" if File.exists?("#{::RbConfig::CONFIG['sysconfdir']}/kafo/kafo.yaml")
       File.join(Dir.pwd, 'config', 'kafo.yaml')
     end
 
+    def select_scenario(config_dir)
+      # try scenario provided in the args via -S or --scenario
+      parsed = ARGV.join(" ").match /(--scenario|-S)(\s+|[=]?)(\S+)/
+      if parsed
+        scenario_file = File.join(config_dir, "#{parsed[3]}.yaml")
+        return scenario_file if File.exists?(scenario_file)
+        puts "Scenario (#{scenario_file}) was not found, can not continue"
+        self.class.exit(:unknown_scenario)
+      end
+
+      # try last_scenario.yaml
+      return self.class.prev_config_file if self.class.prev_config_file
+
+      # assume that in CONF_DIR *.yaml that has key :name is scenario definition
+      scenarios = Dir.glob(File.join(CONFIG_DIR, '*.yaml')).inject({}) do |scns, scn_file|
+        content = YAML.load_file(scn_file)
+        scns[scn_file] = content if content.has_key?(:name)
+        scns
+      end
+
+      # try it whene there is single scenario
+      return scenarios.keys.first if scenarios.keys.count == 1
+
+      # let the user select if in interactive mode
+      if (ARGV & ['--interactive', '-i']).any?
+
+        wizard = KafoWizards.wizard(:cli, 'Select installation scenario',
+          :description => "Please select one of the pre-set installation scenarios. You can customize your installtion later during the installtion.")
+        f = wizard.factory
+
+        scenarios.keys.each do |scn|
+          wizard.entries << f.button(scn, :label => "#{scenarios[scn][:name]}: #{scenarios[scn][:description]}", :default => false)
+        end
+        wizard.entries << f.button(:cancel, :label => 'Cancel Installation', :default => false)
+
+        res =  wizard.run
+        if res == :cancel
+          say 'Installation was cancelled by user'
+          self.class.exit(1)
+        end
+        return res
+      end
+
+      # fail
+      puts "Scenario was not selected, can not continue. Use --list-scenarios to list available options."
+      self.class.exit(:unknown_scenario)
+    end
+
+    def self.list_available_scenarios
+      say "Available scenarios"
+      if defined?(CONFIG_DIR) && Dir.exists?(CONFIG_DIR)
+        Dir.glob(File.join(CONFIG_DIR, '*.yaml')).each do |config_file|
```

Comment:
this should be extracted to some method as the same thing is being done in `select_scenario` method